Drivers/Src/CANbus.c

```diff
@@ -3,73 +3,109 @@
 #include "os.h"
 #include "Tasks.h"
 
-static OS_SEM CANMail_Sem4;       // sem4 to count how many sending hardware mailboxes we have left (start at 3)
-static OS_SEM CANBus_RecieveSem4; // sem4 to count how many msgs in our recieving queue
-static OS_MUTEX CANbus_TxMutex;   // mutex to lock tx line
-static OS_MUTEX CANbus_RxMutex;   // mutex to lock Rx line
+static OS_SEM CANMail_Sem4[NUM_CAN];       // sem4 to count how many sending hardware mailboxes we have left (start at 3)
+static OS_SEM CANBus_RecieveSem4[NUM_CAN]; // sem4 to count how many msgs in our recieving queue
+static OS_MUTEX CANbus_TxMutex[NUM_CAN];   // mutex to lock tx line
+static OS_MUTEX CANbus_RxMutex[NUM_CAN];   // mutex to lock Rx line
 
 /**
  * @brief this function will be passed down to the BSP layer to trigger on RX events. Increments the recieve semaphore to signal message in hardware mailbox. Do not access directly.
+ * @param bus The CAN bus to operate on
  */
-void CANbus_RxHandler()
+void CANbus_RxHandler(CAN_t bus)
 {
     OS_ERR err;
-    OSSemPost(&CANBus_RecieveSem4, OS_OPT_POST_1, &err); // increment our queue counter
+    OSSemPost(&(CANBus_RecieveSem4[bus]), OS_OPT_POST_1, &err); // increment our queue counter
     assertOSError(OS_CANDRIVER_LOC,err);
 }
 
 /**
  * @brief this function will be passed down to the BSP layer to trigger on TXend. Releases hold of the mailbox semaphore (Increments it to show mailbox available). Do not access directly.
+ * @param bus The CAN bus to operate on
  */
-void CANbus_TxHandler()
+void CANbus_TxHandler(CAN_t bus)
 {
     OS_ERR err;
-    OSSemPost(&CANMail_Sem4, OS_OPT_POST_1, &err);
+    OSSemPost(&(CANMail_Sem4[bus]), OS_OPT_POST_1, &err);
     assertOSError(OS_CANDRIVER_LOC,err);
 }
 
+//wrapper functions for the interrupt customized for each bus
+void CANbus_TxHandler_1(){
+    CANbus_TxHandler(CAN_1);
+}
+
+void CANbus_TxHandler_3(){
+    CANbus_TxHandler(CAN_3);
+}
+
+void CANbus_RxHandler_1(){
+    CANbus_RxHandler(CAN_1);
+}
+void CANbus_RxHandler_3(){
+    CANbus_RxHandler(CAN_3);
+}
+
 /**
  * @brief   Initializes the CAN system
  * @param   None
  * @return  None
  */
-void CANbus_Init(void)
+void CANbus_Init(CAN_t bus)
 {
     // initialize CAN mailbox semaphore to 3 for the 3 CAN mailboxes that we have
     // initialize tx
     OS_ERR err;
-
-    OSMutexCreate(&CANbus_TxMutex, "CAN TX Lock", &err);
+    
```

Comment:
To match the error checking in other CAN functions, it might be nice to have an ```assert bus == CAN_1 or bus == CAN_2``` here